File: app/src/main/java/com/example/project_note/DataBase/NoteDAO.java
package com.example.project_note.DataBase;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;
@Dao
public interface NoteDAO {

    @Insert
    void insertNote(Note note);

    @Update
    void updateNote(Note note);
    @Delete
    void deleteNote(Note note);
    @Query("SELECT *FROM note")
    List<Note> getListNote();
    @Query("SELECT *FROM note where title = :title and detail= :detail")
    int getID(String title, String detail);
//    @Query("SELECT *FROM note where  id = :id)
//    int getNote(int id);
}

File: app/src/main/java/com/example/project_note/DataBase/NoteDAO.kt
package com.example.project_note.DataBase

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Transaction
import androidx.room.Update
import io.reactivex.Completable
import io.reactivex.Single

@Dao
interface NoteDAO {
    @Insert
    fun insertNote(note: Note) : Completable

    @Update
    fun updateNote(note: Note) : Completable
    @Delete
    fun deleteNote(note: Note) : Completable

    @Query("SELECT * FROM Note")
    fun getListNote(): MutableList<Note>

    @Query("SELECT id_Note FROM Note WHERE title = :title AND detail= :detail")
    fun getID(title: String, detail: String): Int
}